### ISSUE TITLE
No documentation on how to point a Kubernetes pod to a registry other than gcr.io

### ISSUE THREAD

**Issue by andvikram** [kubernetes/kubernetes#15935]
I have been trying to a run a very basic container based on ubuntu:14.04 that simply echos 'Hello World'. But when I run the kubelet with most of the default options I get this error message:

`Failed to create pod infra container: image pull failed for gcr.io/google_containers/pause:0.8.0, this may be because there are no credentials on this request`

If gcr.io requires credentials how can I point it to use Docker Hub, or more preferably, a local registry? I tried using the --pod-infra-container-image option but it does not seems to work.

**Comment by jfbibeau** [comment]
This is what I did:

`docker pull docker.io/kubernetes/pause`

`docker tag docker.io/kubernetes/pause gcr.io/google_containers/pause:0.8.0`

`KUBELET_ARGS="--pod_infra_container_image=docker.io/kubernetes/pause:latest"`

Restart kubelet service.

However, I think your question is still valid - for example, I want to enable the DNS addon in k8s but I can't reach gcr.io to pull the packages.

**Comment by thockin** [comment]
gcr.io should not require creds for things like 'pause'

@mattmoore

**Comment by jingxiaolu** [comment]
@thockin for 'pause', gcr.io should not requires creds, but what if we deploy docker image 'pause' in a local registry?

We've deployed a local registry, and pushed k8s apps for internal use.
For K8s apps, we can specify [ImagePullSecrets](http://kubernetes.io/docs/user-guide/images/#specifying-imagepullsecrets-on-a-pod) for pulling docker images.
However for the pod infra container image 'pause', we specified it with --pod-infra-container-image for kubelet. When Kubelet trys to pull it from local registry, it can't handle the registry auth. 

So the current workaround we're using is adding `docker pull xxx.com/pause` for every k8s node when it is deployed.

Is it possible you add an option in kubelet, so we can specify 'registrykey', just like what we do in [ImagePullSecrets](http://kubernetes.io/docs/user-guide/images/#specifying-imagepullsecrets-on-a-pod).

Many thanks :-)
